### PR TITLE
[TRACING-4026] Replace tracegen by telemetrygen

### PR DESCRIPTION
**Pull request by iblancasa**
Version(s):   4.16, 4.15, 4.14, 4.13, 4.12

Issue: [TRACING-4026](https://issues.redhat.com//browse/TRACING-4026)

Link to docs preview: https://74350--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-forwarding.html

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by iblancasa** [comment]
Ping @IshwarKanse @max-cx

**Review by IshwarKanse — APPROVED** [review]

**Review by max-cx — APPROVED** [review]
LGTM

File: observability/otel/otel-forwarding.adoc [hunk 1]
```diff
@@ -115,26 +115,25 @@ spec:
 
 [TIP]
 ====
-You can deploy `tracegen` as a test:
+You can deploy `telemetrygen` as a test:
```
Comment (adellape) [inline observability/otel/otel-forwarding.adoc:118]: Suggestion:
```suggestion
You can deploy a `telemetrygen` job as a test:
```

**Comment by skopacz1** [comment]
Hi @iblancasa, I see that your PR has two commits. Our PRs must have only one commit by the time they're ready to be merged, could you please squash them into one commit?

When you're done, please add the merge-review-needed label, or if you finish squashing within the week, feel free to reach out to me and I can proceed with the review. Let me know if you have any questions, thanks!

**Comment by skopacz1** [comment]
sorry @iblancasa it seems that prow issues are causing checks to fail for now. I'll proceed with reviewing this PR in the meantime and provide feedback if there is any. If everything looks good, I'll try to retest throughout the day until checks work so I can get this PR merged.

**Review by skopacz1 — APPROVED** [review]
Merge review LGTM! I'll try retesting throughout the day until the checks pass, then I'll merge

**Pull request merged**